circular-buffer/tests/circular-buffer.rs

```diff
@@ -0,0 +1,79 @@
+extern crate circular_buffer;
+
+#[allow(unused_must_use)]
+mod tests {
+
+    use circular_buffer::{CircularBuffer, Error};
+
+    #[test]
+    fn error_on_read_empty_buffer() {
+        let mut buffer = CircularBuffer::new(1);
+        assert_eq!(Err(Error::EmptyBuffer), buffer.read());
+    }
+
+    #[test]
+    #[ignore]
+    fn write_and_read_back_item() {
+        let mut buffer = CircularBuffer::new(1);
+        buffer.write('1');
+        assert_eq!('1', buffer.read().unwrap());
+        assert_eq!(Err(Error::EmptyBuffer), buffer.read());
+    }
+
+    #[test]
+    #[ignore]
+    fn write_and_read_back_multiple_items() {
+        let mut buffer = CircularBuffer::new(2);
+        buffer.write('1');
+        buffer.write('2');
+        assert_eq!('1', buffer.read().unwrap());
+        assert_eq!('2', buffer.read().unwrap());
+        assert_eq!(Err(Error::EmptyBuffer), buffer.read());
+    }
+
+    #[test]
+    #[ignore]
+    fn alternate_write_and_read() {
+        let mut buffer = CircularBuffer::new(2);
+        buffer.write('1');
+        assert_eq!('1', buffer.read().unwrap());
+        buffer.write('2');
+        assert_eq!('2', buffer.read().unwrap());
+    }
+
+    #[test]
+    #[ignore]
+    fn clear_buffer() {
+        let mut buffer = CircularBuffer::new(3);
+        buffer.write('1');
+        buffer.write('2');
+        buffer.write('3');
+        buffer.clear();
+        assert_eq!(Err(Error::EmptyBuffer), buffer.read());
+        buffer.write('1');
+        buffer.write('2');
+        assert_eq!('1', buffer.read().unwrap());
+        buffer.write('3');
+        assert_eq!('2', buffer.read().unwrap());
+    }
+
+    #[test]
+    #[ignore]
+    fn full_buffer_error() {
+        let mut buffer = CircularBuffer::new(2);
+        buffer.write('1');
+        buffer.write('2');
+        assert_eq!(Err(Error::FullBuffer), buffer.write('3'));
+    }
+
+    #[test]
+    #[ignore]
+    fn overwrite_item_in_full_buffer() {
```

Comment:
Maybe could add a test for using overwrite in a non-full buffer too (like in Ruby) - create with size 2, write 1, overwrite 2, read should return 1 instead of 2